### PR TITLE
Warning fixes

### PR DESCRIPTION
**Pull request by EVAST9919**

### CODE REVIEW

File: osu.Game/OsuGame.cs [hunk 1]
```diff
@@ -214,7 +214,7 @@ private bool globalHotkeyPressed(InputState state, KeyDownEventArgs args)
                 }
             }
 
-            return base.OnKeyDown(state, args);
+            return OnKeyDown(state, args);
```
Comment (Tom94) [inline osu.Game/OsuGame.cs]: Interesting that `OnKeyDown` is triggered here. This might be a mistake.

Comment (EVAST9919) [inline osu.Game/OsuGame.cs]: should I revert it?

Comment (Tom94) [inline osu.Game/OsuGame.cs]: Leave for now. Just something for the next person who can look into it to consider. Just commented so it doesn't get lost.

File: osu.Game/Database/BeatmapDatabase.cs [hunk 2]
```diff
@@ -25,14 +25,12 @@ public class BeatmapDatabase
...
 
-        private BeatmapImporter ipc;
-
         public BeatmapDatabase(Storage storage, GameHost importHost = null)
         {
             this.storage = storage;
 
             if (importHost != null)
-                ipc = new BeatmapImporter(importHost, this);
+                new BeatmapImporter(importHost, this);
```
Comment (peppy) [inline osu.Game/Database/BeatmapDatabase.cs]: if this still actually works, this screams incorrect (unimplemented) disposal logic.

File: osu.Game/OsuGame.cs (hunk 1 above)
Comment (peppy) [inline osu.Game/OsuGame.cs]: does indeed look like an oversight. this should be a `return false`

File: osu.Game/Overlays/Options/OptionDropDown.cs [hunk 3]
```diff
@@ -101,7 +101,7 @@ public OptionDropDown()
                 {
                     Margin = new MarginPadding { Top = 5 },
                     RelativeSizeAxes = Axes.X,
-                    Items = this.Items,
+                    Items = Items,
```
Comment (peppy) [inline osu.Game/Overlays/Options/OptionDropDown.cs]: scary naming

File: osu.Game/Database/BeatmapDatabase.cs (hunk 2 above)
Comment (EVAST9919) [inline osu.Game/Database/BeatmapDatabase.cs]: Beatmaps import works

Comment (peppy) [inline osu.Game/Database/BeatmapDatabase.cs]: that means that disposal/finalizer logic is not implemented, which is an issue. this reference should remain and disposal logic should be fixed.

File: osu.Game/Overlays/Options/OptionDropDown.cs (hunk 3 above)
Comment (EVAST9919) [inline osu.Game/Overlays/Options/OptionDropDown.cs]: Shold I change it back or rename somehow?

**Pull request merged**